### PR TITLE
Refactor samples v1

### PR DESCRIPTION
**Pull request by ramya0820**
## Description

- Refactored to
      ___ examples
      ______ getting started
         ______ advanced features
          ______ to be deleted // Contains older samples for reference and review
               
- Added summaries for the examples
- Made examples queue/topics agnostic and added comments
- Added in notes to re-use sendMessages sample
- Fixed/uniformized error handling, message attributes in use, and few other clean ups as discussed on #43 and in person

- Testing:- Individually executed samples, found and filed bug reports

# Reference to any github issues
- #43

### CODE REVIEW

**Comment by ramya-rao-a** [comment]
@ramya0820 The linter is failing in the travis tests. Can you take a look at them?

image

To get these errors locally, run `npm run tslint`

File: examples/gettingStarted/browseMessages.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  } finally {
+    await ns.close();
+  }
+}
+
+async function browseMessages(): Promise<void> {
+  const client = ns.createQueueClient(queueName); // Use this API to peek from a queue
+  // const client = ns.createSubscriptionClient(topicName, subscriptionName); // Use this API to peek from a topic subscription
+
+  const messages = await client.peek(10); // Peeks one message
```
Comment (bterlson) [inline examples/gettingStarted/browseMessages.ts]: One message? (Parameter is 10...)

File: examples/gettingStarted/scheduledMessages.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    { name: "Newton", firstName: "Isaac" },
+    { name: "Bohr", firstName: "Niels" },
+    { name: "Faraday", firstName: "Michael" },
+    { name: "Galilei", firstName: "Galileo" },
+    { name: "Kepler", firstName: "Johannes" },
+    { name: "Kopernikus", firstName: "Nikolaus" }
+  ];
+
+  for (let index = 0; index < NUM_OF_MESSAGES; index++) {
+    const element = data[index];
```
Comment (bterlson) [inline examples/gettingStarted/scheduledMessages.ts]: s/element/scientist, maybe? Element is a bit overloaded in scientific contexts :)

File: examples/advancedFeatures/deferral.ts [hunk 3]
```diff
@@ -4,63 +4,81 @@ import {
...
+    const promises = new Array();
+    for (let index = 0; index < data.length; index++) {
+      const message: SendableMessageInfo = {
+        body: data[index],
+        label: "RecipeStep",
+        contentType: "application/json"
+      };
+      // the way we shuffle the message order is to introduce a tiny random delay before each of the messages is sent
+      promises.push(
+        delay(Math.random() * 30).then(async () => {
```
[the remark this answers is not in the document]
Comment (ramya0820) [reply examples/advancedFeatures/deferral.ts]: Agree, updated code in newer revision

File: examples/advancedFeatures/deferral.ts [hunk 4]
```diff
@@ -4,63 +4,81 @@ import {
...
+/*
+  This sample demonstrates how Service Message deferral feature can be used to defer messages to be
+  available for receiving by client at a later time.
+
+  We take for example an application receives cooking instructions that the clients need to process
+  in appropriate sequence. We simulate a common scenario where the messages are not received in
+  expected sequence. The example shows how message deferral can be used to process the messages
+  in expected sequence.
+*/
 async function main(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (ramya0820) [reply examples/advancedFeatures/deferral.ts]: Yes, we can discuss more on this. Ramya A. and I had discussed offline about other possible changes related to how we might execute the samples

File: examples/gettingStarted/browseMessages.ts (hunk 1 above)
Comment (ramya0820) [reply examples/gettingStarted/browseMessages.ts]: oops, fixed it!

File: examples/gettingStarted/receiveMessagesStreaming.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    console.log(`Received message: ${brokeredMessage.body} - ${brokeredMessage.label}`);
+
+    await brokeredMessage.complete();
+  };
+  const onError: OnError = (err: MessagingError | Error) => {
+    console.log("Error occurred: ", err);
+  };
+
+  const receiveHandler = client.receive(onMessage, onError, { autoComplete: false });
+  await delay(receiveClienTimeoutInMilliseconds);
```
[the remark this answers is not in the document]
Comment (ramya0820) [reply examples/gettingStarted/receiveMessagesStreaming.ts]: Added a comment for just the Streaming Receive sample. Not sure if we need that for all the samples as such?

File: examples/gettingStarted/scheduledMessages.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    };
+    const scheduledEnqueueTimeUtc = new Date(Date.now() + 30000 + index * 1000); // scheduling message to be sent (30 + index) seconds from now
+    console.log(
+      `>>>> Sending message:\t ${message.body}, scheduled for UTC: ${scheduledEnqueueTimeUtc}`
+    );
+    await client.scheduleMessage(message, scheduledEnqueueTimeUtc);
+  }
+}
+
+async function receiveMessages(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (ramya0820) [reply examples/gettingStarted/scheduledMessages.ts]: I've updated the sample and logged an issue when using with receiveBatch(). Can discuss more offline.

File: examples/gettingStarted/scheduledMessages.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  This sample demonstrates how .scheduleMessage() API can be used to schedule messages to appear
+  on a Service Bus entity at a specified later time.
+*/
+async function main(): Promise<void> {
+  ns = Namespace.createFromConnectionString(connectionString);
+  try {
+    // schedule messages to appear on queue/topic at a later time.
+    await sendScheduledMessages();
+
+    // retrieve all the messages that were sent to the queue
```
[the remark this answers is not in the document]
Comment (ramya0820) [reply examples/gettingStarted/scheduledMessages.ts]: Agree (As with other comments). A little unsure about jsdoc format. Hopefully updated changes are as intended/expected.

File: examples/gettingStarted/receiveMessagesStreaming.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (ramya-rao-a) [reply examples/gettingStarted/receiveMessagesStreaming.ts]: Yes, lets just add here.

**Pull request closed without merging**